tests/cloud_answer_tests_runner.py

```diff
@@ -0,0 +1,390 @@
+
+    Parameters
+    ----------
+    nose_xml : string
+        full path of xml file to be parsed
+
+    Returns
+    -------
+    tuple : (failed_answers, missing_answers)
+
+        failed_answers : list of tuples (string, dict)
+        Collection of tuples where the first part is a string denoting the
+        test name, the second part is a dictionary that stores the actual,
+        expected and difference plot file locations of the test.
+        eg. [('yt.visualization.tests.test_line_plots:test_line_plot',
+                {'Actual': '/usr/tmp/tmp43la9b0w.png',
+                'Expected': '/usr/tmp/tmpbpaqbgi3.png',
+                'Difference': '/usr/tmp/tmp43la9b0w-failed-diff.png'}
+            )]
+
+        missing_answers : list of string
+        Collection of missing answer tests specifying full name of the test.
+        eg. ['yt.visualization.tests.test_line_plots:test_multi_line_plot']
+
+    """
+    missing_answers = []
+    failed_answers = []
+
+    tree = ET.parse(nose_xml)
+    testsuite = tree.getroot()
+
+    for testcase in testsuite:
+        for error in testcase:
+            test_name = testcase.attrib["classname"] + ":" \
+                        + testcase.attrib["name"]
+            if "No such file or directory" in error.attrib["message"]:
+                missing_answers.append(test_name)
+            elif "Items are not equal" in error.attrib["message"]:
+                img_path = extract_image_locations(error.attrib["message"])
+                if img_path:
+                    failed_answers.append((test_name, img_path))
+    return failed_answers, missing_answers
+
+if __name__ == "__main__":
+    """Run answer tests on cloud platforms like Travis, Appveyor
+
+    This script executes yt answer tests on cloud platform.
+    If the test fail due to difference in actual and expected images, 
+    this function uploads a html page having all the plots which got failed 
+    (if executed with `-f` command line argument).
+    In case, answer store does not has a golden answer and if executed with 
+    `-m` argument, it uploads missing answers zip file to yt's curldrop server.
+    
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-f", "--upload-failed-tests", action="store_true",
+                        help="Upload a comparison report of failed answer tests"
+                             " to yt's curldrop server.")
+    parser.add_argument("-m", "--upload-missing-answers", action="store_true",
+                        help="Upload tests' answers that are not found in "
+                             "answer-store.")
+    args = parser.parse_args()
+
+    # ANSI color codes
+    COLOR_BLUE = '\x1b[34;1m'
+    COLOR_CYAN = '\x1b[36;1m'
+    COLOR_RESET = '\x1b[0m'
+
+    ANSWER_STORE = ytcfg.get("yt", "test_storage_dir")
+    if not ANSWER_STORE or ANSWER_STORE == "/does/not/exist":
+        ANSWER_STORE = "answer-store"
+
+    NOSETEST_XML = "answer_testing_nosetests.xml"
```

Comment:
Super minor, but it doesn't make that much sense to have a "constant" here and then pass it as the argument to `parse_nose_xml`. I'd just have the constant there.